magicctapipe/scripts/lst1_magic/semi_automatic_scripts/check_MAGIC_runs.py

```diff
@@ -0,0 +1,86 @@
+#This script allows us to get information about every MAGIC run ID (and subruns) in files used for common data analysis (MAGIC1, MAGIC2, LST1). 
+
+#The MAGIC files that can be used for analysis are located in the IT cluster in the following directory:
+#/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}
+
+#In this path, 'tel_id' refers to the telescope ID, which must be either 1 or 2. 'YYYY', 'MM', and 'DD' specify the date.
+
+#In the first step, we have to load a dataframe that contains information about the date, the name of the source, and the range of MAGIC #runs. The file in file_path was generated using the spreadsheet "Common MAGIC LST1 data".
+
+import pandas as pd
+from datetime import datetime, timedelta
+import os
+import re
+
+file_path = '/fefs/aswg/workspace/joanna.wojtowicz/data/magic_first_and_last_runs.csv'
+df = pd.read_csv(file_path,sep='\t', dtype={'Date (LST conv.)': str, 'Source': str, 'First run': int, 'Last run': int})
+
+def check_run_ID(path, filename, first_run, last_run, date, source):
+    Y = f'_Y_{source}' 
+    #'Y' because we have to be sure that the function counts right filename.
+
+    if Y in filename:
+        # Extract numbers from filename and check range
+        run_ids = [int(s) for s in re.findall(r'\d+', filename)]
```

Comment:
I still do not like how the run_ids are extracted: since you get all the numbers in the file name: date, numerical parts of the source name, subrun, wobble angles, etc., it might also happen that you get runid-like number in one of those other numbers.
Is filename the base name, or with the full path?
If it is the base name you can extract the run id (and only this):
```int(filename.split("_")[2].split(".")[0])```